### ISSUE TITLE
for_update - unexpected token IDENTIFIER(FOR)

### ISSUE THREAD

**Issue by wangzj** [phalcon/cphalcon#2811]
I've tried to use 'for_update' as the following way:

    <?php
    Models\Account::findFirst(array("passport='$passport'", "for_update" => true));

This reports an error, like

> Syntax error, unexpected token IDENTIFIER(FOR), near to ' UPDATE', when parsing: SELECT [Models\Account].* FROM [Models\Account] WHERE passport='...' LIMIT 1 FOR UPDATE (100)

I found that `cphalcon/ext/mvc/model/query/parser.h` contained no 'PHQL_FOR'. Does that matter? Any suggestions?

Phalcon **1.3.3**

**Comment by NeoEGM** [comment]
Same problem here, version 1.3.4:

```
Syntax error, unexpected token IDENTIFIER(FOR), near to ' UPDATE', when parsing: SELECT [XXX].* FROM [XXX] WHERE enabled = 1 AND activated = 1 AND id != ?1 ORDER BY assigned_count ASC, id ASC LIMIT 5 FOR UPDATE (189)
```

Query code:

```php
$users = Users::find([
    			"conditions" => "enabled = 1 AND activated = 1 AND id != ?1",
    			"bind" => [1 => $this->id],
    			"order" => "assigned_count ASC, id ASC",
    			"limit" => 5,
    			'for_update' => true
    	]);
```

**Comment by geekzgit** [comment]
me too

**Comment by lixiphp** [comment]
me too

**Comment by hugofcampos** [comment]
Is this still up for fixes? Is anybody looking for this?

**Comment by andresgutierrez** [comment]
Implemented in Phalcon 2.0.4
